Source/Core/DolphinWX/Debugger/CodeWindow.cpp

```diff
@@ -316,34 +314,40 @@ void CCodeWindow::StepOver()
       PowerPC::breakpoints.ClearAllTemporary();
       PowerPC::breakpoints.Add(PC + 4, true);
       CPU::EnableStepping(false);
-      JumpToAddress(PC);
-      Update();
     }
     else
     {
       SingleStep();
     }
-
-    UpdateButtonStates();
-    // Update all toolbars in the aui manager
-    Parent->UpdateGUI();
   }
 }
 
+// Returns true on a blr or on a bclr that evaluates to true.
+static bool WillInstructionReturn(UGeckoInstruction inst)
+{
+  bool counter = (inst.BO_2 >> 2) != 0 || (CTR != 0) != ((inst.BO_2 >> 1) != 0);
```

Comment:
Does it work with others conditional branching instructions?
> bcx (According to PPC manual)
> ctr_ok ← BO[2] | ((CTR ≠ 0) ⊕ BO[3])
> cond_ok ← BO[0] | (CR[BI] ≡ BO[1])

Shouldn't it be like this:
```c++
bool counter = (inst.BO_2 >> 2 & 1) | (CTR != 0) ^ (inst.BO_2 >> 1 & 1)
```
In a less binary way:
```c++
bool counter = (inst.BO_2 >> 2 & 1) != 0 || (CTR != 0) != ((inst.BO_2 >> 1 & 1) != 0);
```